kintoclient/__init__.py

```diff
@@ -0,0 +1,247 @@
+            The bucket object which owns this collection.
+        :param session:
+            An optional session object to use, rather than creating a new one.
+        :param create:
+            Defines if the collection should be created. (default to False)
+        """
+        self.session = create_session(server_url, auth, session)
+        if isinstance(bucket, six.string_types):
+            bucket = Bucket(bucket, session=session)
+        self.bucket = bucket
+        self.name = name
+        self.uri = "%s/collections/%s" % (self.bucket.uri, self.name)
+
+        method = 'put' if create and self.bucket.name != 'default' else 'get'
+        request_kwargs = {}
+        if method == 'put' and permissions is not None:
+            request_kwargs['permissions'] = permissions
+
+        info, _ = self.session.request(method, self.uri, **request_kwargs)
+        self.data = info['data']
+        self.permissions = Permissions('collection', info['permissions'])
+
+    def _get_record_uri(self, record_id):
+        return '%s/records/%s' % (self.uri, record_id)
+
+    def get_records(self):
+        """Returns all the records"""
+        # XXX Add filter and sorting.
+        records_uri = '%s/records' % self.uri
+        resp, _ = self.session.request('get', records_uri)
+
+        # XXX Support permissions for GET /records.
+        return [Record(data, collection=self) for data in resp['data']]
+
+    def get_record(self, id):
+        record_uri = self._get_record_uri(id)
+        resp, _ = self.session.request('get', record_uri)
+        return Record(resp['data'],
+                      collection=self,
+                      permissions=resp['permissions'])
+
+    def save_record(self, record):
+        if record.id is None:
+            record.id = str(uuid.uuid4())
+        self.session.request('put', self._get_record_uri(record.id))
+
+    def save_records(self, records):
+        # XXX enhance this with a batch request.
+        for record in records:
+            self.save_record(record)
+
+    def create_record(self, data, permissions=None, save=True):
+        record = Record(data, permissions=permissions, collection=self)
+        if save is True:
+            self.save_record(record)
+        return record
+
+    def delete_record(self, id):
+        record_uri = self._get_record_uri(id)
+        resp, _ = self.session.request('delete', record_uri)
+
+    def delete_records(self, records):
+        # XXX TODO
+        # with self.session.batch() as batch:
+        #     for record in records:
+        #         batch.request('delete', self._get_record_uri(record.id))
+        pass
+
+
+class Record(object):
+    """Represents a record"""
+
+    def __init__(self, data, collection, permissions=None, id=None):
+        if id is None:
+            id = str(uuid.uuid4())
+        self.id = id
+        self.collection = collection
+        self.permissions = Permissions('record', permissions)
+        self.data = data
```

Comment:
`Request` and `Collection` objects actually issue requests themselves, whereas the `Record` waits to be populated by the `Collection.get_record`. I'm not sure which solution I prefer:
1. Always do the request in the constructor;
2. Always do the request in the `get_xxx` method of the parent object;
3. Don't bother and let it like it is right now.